web/modules/weather_cms/weather_cms.module

```diff
@@ -124,3 +126,32 @@ function weather_cms_entity_presave(Drupal\Core\Entity\EntityInterface $entity)
         $entity->set('field_derived_wfo', 'unknown');
     }
 }
+
+/**
+ * Hook after `filefield_paths_filefield_paths_process_file` runs.
+ */
+function weather_cms_filefield_paths_process_file(
+    ContentEntityInterface $entity,
+    FieldItemListInterface $field,
+    array $settings
+) {
+    // if we have a weather story upload, replace the file in-place if applicable.
+    if ($entity->bundle() === 'wfo_weather_story_upload') {
+        $file_system = \Drupal::service('file_system');
+        foreach ($field->referencedEntities() as $file) {
+            $origname = $file->get('origname')->value;
+            $location = $file->get('uri')->value;
+            $dirname = $file_system->dirname($location);
+            $new_location = "{$dirname}/{$origname}";
+
+            if ($location !== $new_location) {
+                $message = "weather story upload: moving {$location} to {$new_location}";
+                \Drupal::logger("weather_cms")->notice($message);
+                // NB `$file_system->moveUploadedFile` does not work here, so do
+                // the renaming manually.
+                $file_system->delete($new_location);
+                $file_system->move($location, $new_location);
```

Comment:
What do you think of _not_ deleting `$new_location` and instead using `move` with the optional 3rd parameter `$fileExists` as `FileExists::Replace`? See [FileSystem::move](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileSystem.php/function/FileSystem%3A%3Amove/11.x) and [FileExists.php](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileExists.php/11.x).

Reply:
Fantastic idea! Trying this out.

Reply:
Works like a charm. Many thanks!